File: 07.3 ProgressDialogTest/src/net/terzeron/progressdialogtest/MainActivity.java
package net.terzeron.progressdialogtest;

import android.os.Bundle;
import android.app.Activity;
import android.app.ProgressDialog;
import android.view.Menu;

public class MainActivity extends Activity {

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_main);
		
		/*ProgressDialog dialog =*/ ProgressDialog.show(MainActivity.this, "", "Loading. Please wait...", true);
		
		// 이유를 알 수 없지만 제대로 동작하지 않음
//		ProgressDialog progressDialog;
//		progressDialog = new ProgressDialog(MainActivity.this);
//		//progressDialog.setProgressStyle(ProgressDialog.STYLE_HORIZONTAL);
//		progressDialog.setMessage("Loading...");
//		progressDialog.setCancelable(false);
	}

	@Override
	public boolean onCreateOptionsMenu(Menu menu) {
		// Inflate the menu; this adds items to the action bar if it is present.
		getMenuInflater().inflate(R.menu.main, menu);
		return true;
	}

}

File: 07MotionEventTest/app/src/main/java/net/terzeron/a07motioneventtest/MainActivity.kt
package net.terzeron.a07motioneventtest

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.MotionEvent
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        activity_motion_event.setOnTouchListener { v: View, m: MotionEvent ->
            handleTouch(m)
            true
        }
    }

    private fun handleTouch(m: MotionEvent) {
        val pointerCount = m.pointerCount
        for (i in 0 until pointerCount) {
            val x = m.getX(i)
            val y = m.getY(i)
            val id = m.getPointerId(i)
            val action = m.actionMasked
            val actionIndex = m.actionIndex
            val actionString: String

            when (action) {
                MotionEvent.ACTION_DOWN -> actionString = "DOWN"
                MotionEvent.ACTION_UP -> actionString = "UP"
                MotionEvent.ACTION_POINTER_DOWN -> actionString = "POINTER_DOWN"
                MotionEvent.ACTION_POINTER_UP -> actionString = "POINTER_UP"
                MotionEvent.ACTION_MOVE -> actionString = "MOVE"
                else -> actionString = ""
            }

            val touchStatus = "Action: $actionString Index: $actionIndex ID: $id X: $x, Y: $y"
            if (id == 0) {
                textView1.text = touchStatus
            } else {
                textView2.text = touchStatus
            }
        }
    }
}
